File: server/tests/fixtures/alpha_rule_triggers/java/sec_sql_injection_concat.java
// Should trigger: sec.sql_injection_concat
import java.sql.*;

public class SqlInjection {
    public void queryUser(Connection conn, String userId) throws SQLException {
        String query = "SELECT * FROM users WHERE id = " + userId;  // SQL injection!
        Statement stmt = conn.createStatement();
        stmt.executeQuery(query);
    }
}

File: server/tests/fixtures/alpha_rule_triggers/python/sec_sql_injection_concat.py
# Should trigger: sec.sql_injection_concat
import sqlite3

def unsafe_query(user_input):
    conn = sqlite3.connect(':memory:')
    cursor = conn.cursor()
    query = "SELECT * FROM users WHERE name = '" + user_input + "'"
    cursor.execute(query)
    return cursor.fetchall()
